packages/password-input/src/InputIcon/InputIcon.styles.ts

```diff
@@ -0,0 +1,52 @@
+import { css } from '@leafygreen-ui/emotion';
+import { Theme } from '@leafygreen-ui/lib';
+import { palette } from '@leafygreen-ui/palette';
+import { spacing } from '@leafygreen-ui/tokens';
+
+import { SizeVariant, States } from '../PasswordInput/PasswordInput.types';
+
+import { StateProps } from './InputIcon.types';
+
+export const baseStyles = css`
+  position: absolute;
+  top: 50%;
+
+  transform: translate(0%, -50%);
+`;
+
+export const sizeStyles: Record<SizeVariant, string> = {
+  [SizeVariant.Small]: css`
+    right: ${spacing[1] * 7}px;
+  `,
+  [SizeVariant.Default]: css`
+    right: ${spacing[5]}px;
+  `,
+  [SizeVariant.Large]: css`
+    right: ${spacing[1] * 10}px;
+  `,
+};
+
+export const themeStateStyles: Record<Theme, Record<StateProps, string>> = {
```

Comment:
Should icon color change when disabled?

Reply:
it doesn't in `TextInput` but i'll double check with @sooachung

Reply:
it should gray out, i'll update